src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,47 @@
+import Popup from "./Popup.js";
+
+class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  open(cardData) {
+    super.open();
+    this._cardData = cardData;
+  }
+
+  close() {
+    this._popupForm.reset();
+    super.close();
+  }
+
+  setButtonText(submit, buttonText = "Saving...") {
```

Comment:
`"Saving..."`


When you delete a card you need `Deleting...`

Reply:
completed.